hbase-server/src/main/jamon/org/apache/hadoop/hbase/tmpl/master/MasterStatusTmpl.jamon

```diff
@@ -698,17 +699,37 @@ AssignmentManager assignmentManager = master.getAssignmentManager();
         <th></th>
         <th>ServerName</th>
         <th>Stop time</th>
+        <%if !master.isInMaintenanceMode() && master.getMasterCoprocessorHost() != null %>
+            <%if RSGroupUtil.isRSGroupEnabled(master.getConfiguration()) %>
+                <th>RSGroup</th>
+            </%if>
+        </%if>
     </tr>
     <%java>
-       DeadServer deadServerUtil = master.getServerManager().getDeadServers();
-       ServerName [] deadServerNames = deadServers.toArray(new ServerName[deadServers.size()]);
-         Arrays.sort(deadServerNames);
-         for (ServerName deadServerName: deadServerNames) {
+        RSGroupInfoManager inMgr = null;
+        DeadServer deadServerUtil = master.getServerManager().getDeadServers();
+        ServerName [] deadServerNames = deadServers.toArray(new ServerName[deadServers.size()]);
+        Arrays.sort(deadServerNames);
+        if (!master.isInMaintenanceMode() && master.getMasterCoprocessorHost() != null
+            && RSGroupUtil.isRSGroupEnabled(master.getConfiguration())) {
+                inMgr = master.getRSGroupInfoManager();
+        }
+        for (ServerName deadServerName: deadServerNames) {
+            String rsGroupName = "default";
+            if (inMgr != null){
+                RSGroupInfo groupInfo = inMgr.getRSGroupOfServer(deadServerName.getAddress());
+                rsGroupName = (groupInfo == null ? "default" : groupInfo.getName());
+            }
     </%java>
     <tr>
     	<th></th>
         <td><% deadServerName %></td>
         <td><% deadServerUtil.getTimeOfDeath(deadServerName) %></td>
+        <%if !master.isInMaintenanceMode() && master.getMasterCoprocessorHost() != null %>
```

Comment:
Set the rsGroupName be null as default, and here use condition if(rsGroupName!=null)